spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitAdmin.java

```diff
@@ -589,11 +589,20 @@ private void declareBindings(final Channel channel, final Binding... bindings) t
 			this.applicationEventPublisher.publishEvent(event);
 		}
 		if (this.ignoreDeclarationExceptions || (element != null && element.isIgnoreDeclarationExceptions())) {
-			if (this.logger.isWarnEnabled()) {
-				this.logger.warn("Failed to declare " + elementType
+			if (this.logger.isDebugEnabled()) {
+				this.logger.debug("Failed to declare " + elementType
 						+ (element == null ? "broker-generated" : ": " + element)
 						+ ", continuing...", t);
 			}
+			else if (this.logger.isWarnEnabled()) {
+				Throwable cause = t;
+				if (t instanceof IOException && t.getCause() != null) {
+					cause = t.getCause();
+				}
+				this.logger.warn("Failed to declare " + elementType
+						+ (element == null ? "broker-generated" : ": " + element)
```

Comment:
M-m-m.
Shouldn't we place `: ` before `broker-generated`? Otherwise it looks like:
```
"Failed to declare queuebroker-generated, continuing... 
```